.github/workflows/lab-validation-pr.yml

```diff
@@ -0,0 +1,172 @@
+name: Lab Validation for PR
+
+on:
+  # Trigger when label is added to PR or new commits are pushed (if label exists)
+  pull_request:
+    types: [labeled, synchronize]
+
+jobs:
+  # Check if we should run based on trigger type
+  check-trigger:
+    runs-on: ubuntu-latest
+    outputs:
+      should_run: ${{ steps.check.outputs.should_run }}
+      pr_number: ${{ steps.check.outputs.pr_number }}
+    steps:
+    - name: Check trigger conditions
+      id: check
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        SHOULD_RUN="false"
+        PR_NUMBER=""
+        
+        if [ "${{ github.event.action }}" = "labeled" ]; then
+          # Check if 'lab-validation' label was added
+          if [ "${{ github.event.label.name }}" = "lab-validation" ]; then
+            SHOULD_RUN="true"
+            PR_NUMBER="${{ github.event.number }}"
+            echo "🏷️ Label 'lab-validation' added to PR #$PR_NUMBER"
+          fi
+        elif [ "${{ github.event.action }}" = "synchronize" ]; then
+          # Check if PR has 'lab-validation' label when new commits are pushed
+          LABELS=$(gh api repos/batfish/batfish/pulls/${{ github.event.number }}/labels --jq '.[].name')
+          if echo "$LABELS" | grep -q "lab-validation"; then
+            SHOULD_RUN="true"
+            PR_NUMBER="${{ github.event.number }}"
+            echo "🔄 New commits pushed to PR #$PR_NUMBER with 'lab-validation' label"
+          fi
+        fi
+        
+        echo "should_run=$SHOULD_RUN" >> $GITHUB_OUTPUT
+        echo "pr_number=$PR_NUMBER" >> $GITHUB_OUTPUT
+
+  # Build Batfish JAR from the PR branch
+  build-batfish:
+    runs-on: ubuntu-latest
+    needs: check-trigger
+    if: needs.check-trigger.outputs.should_run == 'true'
+    outputs:
+      pr_ref: ${{ steps.get_pr.outputs.pr_ref }}
+      pr_title: ${{ steps.get_pr.outputs.pr_title }}
```

Comment:
Is none of this in the input event? We have to actually pull to from the API?

Reply:
The PR title and branch ref aren't available in 'labeled' events (which only contain the label that was added). We need the API call to get this metadata for consistent messaging across both trigger types. This is also standard practice in GitHub workflows that need PR details beyond what's in the triggering event.